torch/utils/_python_dispatch.py

```diff
@@ -252,12 +263,20 @@ def alias_non_inplace_storage(arg, ret):
                 # Example: out = inp.expand(inp.shape[0], inp.shape[0])
                 #     This requires swapping the storage of out to be the same as inp,
                 #     but we do *not* want it to change the sizes/strides that were compute for out.
-                if isinstance(ret, list):
-                    for r in ret:
-                        torch.ops.aten.set_.source_Storage_storage_offset(r, arg.untyped_storage(), r.storage_offset(), r.shape)
-                else:
-                    assert isinstance(ret, torch.Tensor), f"type: {type(ret)}"
-                    torch.ops.aten.set_.source_Storage_storage_offset(ret, arg.untyped_storage(), ret.storage_offset(), ret.shape)
+
+                # Don't dispatch to predispathc
+                with torch._C._ForceDispatchKeyGuard(
+                    torch._C._dispatch_tls_local_include_set(),
+                    torch._C._dispatch_tls_local_exclude_set().add(torch._C.DispatchKey.PreDispatch)
```

Comment:
Hmm. If you look a few lines up, we're also using `with torch.utils._mode_utils.no_dispatch()`. That's implemented [here](https://github.com/pytorch/pytorch/blob/main/torch/csrc/utils/disable_torch_function.h#L23), and it also manually adds the Python key (and related keys) to the exclude set. I think we should just update that API to also automatically exclude PreDispatch, instead of needing an extra guard here.

Reply:
Good catch! I was wondering why it was so asymmetric lol